src/rebar_packages.erl

```diff
@@ -254,6 +254,13 @@ update_package(Name, RepoConfig=#{name := Repo}, State) ->
             fail
     end.
 
+get_package_repo_config(RepoConfig=#{mirror_of := Repo}) ->
```

Comment:
We need to add the mirror_of repo config because hex core uses the name of the repository to verify its authenticity.

Reply:
... that sounds like a bad way to do it?

Why isn't it using keys? Does that mean that anybody who creates a similarly named-repo as a known mirror can sneak in bad data?

Reply:
The registry is signed, but the name of the repository and the package have to be included in the data to be signed. Both the signature and the names need to be verified by the client so that a registry record can only be served for that specific package.

Reply:
Previously only the signature was verified, with this change (including the changes in hex_core) we are also verifying the names. This is to prevent a mirror from serving a signed registry records for the wrong packages.